libopenage/console/console.cpp

```diff
@@ -81,13 +81,26 @@ void Console::register_to_engine(Engine *engine) {
 		return true;
 	});
 	this->input_context.bind(input::event_class::NONPRINT, [this](const input::action_arg_t &arg) {
-		if (arg.e.as_char() == 13) {
-			this->buf.write('\n');
-			this->interpret(this->command);
-			this->command = "";
-			return true;
-		}
-		return false;
+		switch (arg.e.as_char()) {
+			case 8: // remove a single UTF-8 character
+				if (command.size() > 0) {
+					// remove all the extra UTF-8 bytes
+					while((command.back() & 0x80) && !(command.back() & 0x40))
+						command.pop_back();
```

Comment:
this is kinda redundant with the buf utf8 char dropping. can you unify that? maybe even put a function in `util/unicode.h` so any unicode string could benefit from this deletion extension?